### ISSUE TITLE
[Multiple Datasource] [Plugins Version Decoupling] Fetch and include Data Source Engine Type in the data-source saved object when creating a new data source in MDS

### ISSUE THREAD

**Issue by ZilongX** [opensearch-project/OpenSearch-Dashboards#7021]
**Is your feature request related to a problem? Please describe.**

This feature is required per discussion and feedback collected from RFC https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5877.

In order to decouple the plugin versions with OS cluster versions, we will need to specify a supported data source version range for example `>= 2.1.0`, yet given we support both OpenSearch and ElasticSearch clusters, the version could be overlapped since we only record the numeric part, so for an ES 7.10.2 data source the version would be recorded as `7.10.2`, which would break the support range check as it would also satisfy the semver check of `>=2.1.0`

**Describe the solution you'd like**

Add a new layer of `dataSourceType`, to record the backend engine type like `OpenSearch` or `ElasticSearch`

**Describe alternatives you've considered**

Basically this approach would be the most straightforward and simplified one to resolve the issue.

**Additional context**

[Plugins Version Decoupling] Dev work is in progress and targeting release `2.16.0`
